### PR TITLE
 Duplicate Detection: simple list view to show + merge potential duplicates

### LINKED ISSUE #3875: Duplicate Detection: simple list view to show + merge potential duplicates
_As a user I want to see a list of possible duplicate records in order to do a merge with one click, without a lot of external search and analysis._

**Objective**
The "merge" functionality already allows to combine two records into one. But going into bulk action and finding two possible duplicates takes lots of time and is difficult. Offer a special view that already lists all pairs of possible duplicates, including a "merge" button to directly trigger the merge action from there.

**Proposed Solutions & Alternatives**
- [ ] implement a custom component to search and list potential duplicates
- [ ] implement a service to search duplicates
   - for now: very simple exact matching of all selected fields (with trim + lowercase applied to string fields)
- postponed for later:
   - similary scoring
   - "mark not duplicate"
   - integrating configuration of "identifier" fields into the entity schema
   - ... 

Image

### PR DESCRIPTION
**Pull request by Abhinegi2**
- closes #3875

### PR Checklist
_Before you request a manual PR review from someone, please go through all of this list, check the points and mark them checked here.
This helps us reduce the number of review cycles and use the time of our core reviewers more effectively._

- [ ] all automatic CI checks pass (🟢)
  - i.e. code formatting (`ng lint`) passes
  - i.e. unit tests (`ng test`) passes
  - in some cases unit test coverage can be difficult to achieve. If that or any other CI check fails, you can also leave a short comment here why this should be accepted anyway.
- [ ] manually tested (all) required functionality described in the issue  manually yourself on the final version of the code (developer)
- [ ] reviewed the "Files changed" section of the PR briefly yourself to check for any unwanted changes
   - e.g. clean up debugging console.log statements, disabled tests, ...
   - please also avoid changes that are not directly related to the issue of the PR, even small code reformatings make the review process much more complex
- [ ] 🚦 PR status is changed to "Ready for Review"
  - while you are still working on initial implementation, keep the PR in "Draft" status
  - once you are done with your initial work, change to "Ready for Review". This will trigger some additional automated checks and reviews.
  - (PR = "Ready for Review" does not immediately request a manual review yet, first complete the further checks below)
- [ ] marked each code review comment as resolved OR commented on it with a question, if unsure
  - both implementing suggestions of automatic code review or discarding them as not applicable is okay
- [ ] all checkboxes in this checklist are checked (to show the reviewer this really is ready)
- [ ] 🚦 moved the issue related to this PR to Status "Functional Review" to request a personal review

⏪ if issues are commented from testing or code review, make changes. Please then re-check every item of the checklist here again.

-----
### Remaining TODOs:
- [ ] unit tests

## Summary by CodeRabbit

* **New Features**
  * Added "Review Possible Duplicates" menu option for non-internal entity types.
  * New "Review Possible Duplicates" page to search by entity type and matching fields, view results, and compare & merge pairs.
  * Table view with pagination and clear/search controls; progress and no-results messaging included.

* **Tests**
  * Added unit tests covering duplicate detection logic and matching edge cases.

### CODE REVIEW

**Comment by sleidig** [comment]
@Abhinegi2 functionality looks great! :heart_eyes: 

Please go ahead and finalize the code, if necessary.

File: src/app/features/de-duplication/duplicate-detection.service.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+          !usedIds.has(entities[j].getId()) &&
+          this.allFieldsMatch(entities[i], entities[j], fields)
+        ) {
+          pairs.push({ record: entities[i], possibleDuplicate: entities[j] });
+          usedIds.add(entities[i].getId());
+          usedIds.add(entities[j].getId());
+          break;
+        }
+      }
+    }
```
[the remark this answers is not in the document]
Comment (Abhinegi2) [reply src/app/features/de-duplication/duplicate-detection.service.ts:41]: This is the first version for now, we plan to use it as is and revisit improvements later.

File: src/app/features/de-duplication/add-default-review-duplicates-view.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { ConfigMigration } from "../../core/config/config-migration";
+
+export const addDefaultReviewDuplicatesViewConfig: ConfigMigration = (
```
Comment (sleidig) [inline src/app/features/de-duplication/add-default-review-duplicates-view.ts:3]: I am thinking about feature routing in general a bit here: https://github.com/Aam-Digital/ndb-core/issues/3884

Do you think we should make this feature have a hard-wired module routing, similar to /admin?
Then we would not need this config and also could skip registering the component as a "dynamic" component under the `"ReviewDuplicates"` string, I think.

File: src/app/features/de-duplication/review-duplicates/review-duplicates.component.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  });
+
+  onEntityTypeChange(type: string) {
+    this.searchRequestId++;
+    this.selectedEntityType.set(type);
+    this.selectedFields.set([]);
+    this.pairs.set([]);
+    this.searched.set(false);
+    this.isLoading.set(false);
+    this.pageIndex.set(0);
```
Comment (sleidig) [inline src/app/features/de-duplication/review-duplicates/review-duplicates.component.ts]: is this implicitly a `clear()` that we could call here instead?

File: src/app/features/de-duplication/duplicate-detection.service.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      const valA = this.normalizeValue(a[field]);
+      const valB = this.normalizeValue(b[field]);
+      return valA !== "" && valA === valB;
+    });
+  }
+
+  private normalizeValue(value: unknown): string {
+    if (value == null) return "";
+    if (value instanceof Date) return value.toISOString().toLowerCase();
+    if (Array.isArray(value)) return "";
```
Comment (sleidig) [inline src/app/features/de-duplication/duplicate-detection.service.ts:59]: does that mean arrays are always or never matched currently? (okay for now, I think. But maybe document the behavior as comment?)

**Review by sleidig — CHANGES_REQUESTED** [review]
Looks and works great!

File: src/app/features/de-duplication/add-default-review-duplicates-view.ts (hunk 2 above)
Comment (Abhinegi2) [reply src/app/features/de-duplication/add-default-review-duplicates-view.ts:3]: I reviewed it. its an architectural change and it make sense (hard-wired feature routes + prefixed config routes) and is broader than this PR.
so we can keep current PR scoped, and handle routing refactor in follow-up tied to [issue #3884](https://github.com/Aam-Digital/ndb-core/issues/3884).

**Review by sleidig — APPROVED** [review]
